File: src/CSharp/1.TwoSum.cs

using System;
using System.Linq;
using System.Collections;
using System.Collections.Generic;

namespace TwoSumSolution
{
    public class Solution
    {
        public int[] TwoSum(int[] nums, int target) 
        {      
            var dict = new Dictionary<int, int>();
            for (int i = 0; i < nums.Length; i++)
            {
                if(dict.TryGetValue(target - nums[i], out var j) == true)
                {
                    var result = new int[]{0, 0};
                    result[0] = j;
                    result[1] = i;
                    return result;
                }
                if(dict.ContainsKey(nums[i]) == false)
                {
                    dict.Add(nums[i], i);
                }
            }    
            return null;    
        }

    }
}

File: src/Python/1-100/1.TwoSum.py

class Solution:

    isfind = False

    def twoSum(self, nums, target):
        """
        :type nums: List[int]

        :type target: int

        :rtype: List[int]
        """
        n = len(nums)
        for i in range(n):
            for j in range(i + 1, n):
                if nums[i] + nums[j] == target:
                    return [i, j]

        # recursive call overflow
        # return self.__twoSum(nums, n - 1, target)
   
    def __twoSum(self, nums, index, target):
        if index >= 0 and self.isfind == False:
            result = self.__twoSum(nums, index - 1, target)
            if result != None:
                return result
        for i in range(index):
            self.isfind = (nums[i] + nums[index]) == target
            if self.isfind == True:
                return [i, index]

